pkg/policy/l4.go

```diff
@@ -240,6 +242,31 @@ func (l4 *L4Filter) GetPort() uint16 {
 	return uint16(l4.Port)
 }
 
+// NamedPort represents a mapping from a port name to the port number and protocol
+type NamedPort struct {
+	Proto uint8  // 0 for any
+	Port  uint16 // non-0
+}
+
+type NamedPortsMap map[string]NamedPort
+
+func (npm NamedPortsMap) GetNamedPort(name string, proto uint8) (port uint16, err error) {
+	if npm == nil {
+		return 0, fmt.Errorf("nil map")
+	}
+	np, ok := npm[name]
+	if !ok {
+		return 0, fmt.Errorf("named port not found (%v)", npm)
+	}
+	if np.Proto != 0 && proto != np.Proto {
+		return 0, fmt.Errorf("incompatible proto")
+	}
+	if np.Port == 0 {
```

Comment:
When can this happen? Shouldn't this be rejected during validation earlier?

Reply:
Should not happen, but wanted to be safe as 0 port in policy logic indicates wildcarded L4 and it would be bad if 0 slipped through here.